string now_motion_name
string before_motion_name
